### PR TITLE
Feature/add lwt metrics

### PR DESCRIPTION
**Pull request by RobinAtherton**
**Motivation**

HiveMQ has no metrics for tracking last will testaments.

Resolves <https://github.com/hivemq/hivemq-community-edition/issues/66>

**Changes**

Added
- Metric for clients connecting with a last will testament
- Metric for last will testaments sent out on client disconnect

### CODE REVIEW

File: src/main/java/com/hivemq/metrics/HiveMQMetrics.java [hunk 1]
```diff
@@ -133,5 +129,17 @@
      */
     public static final HiveMQMetric<Gauge<Number>> CLIENT_SESSIONS_CURRENT =
             HiveMQMetric.gaugeValue("com.hivemq.sessions.overall.current");
+
+    /**
+     * represents a {@link Counter}, which measures the count of last will testaments.
+     */
+    public static final HiveMQMetric<Counter> LWT_CURRENT =
+            HiveMQMetric.valueOf("com.hivemq.last-will-testaments.overall.current", Counter.class);
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/metrics/HiveMQMetrics.java]:
```suggestion
            HiveMQMetric.valueOf("com.hivemq.will.count", Counter.class);
```

File: src/main/java/com/hivemq/metrics/HiveMQMetrics.java [hunk 2]
```diff
@@ -133,5 +129,17 @@
...
+
+    /**
+     * represents a {@link Counter}, which measures the count of last will testaments spread to clients.
+     */
+    public static final HiveMQMetric<Counter> LWT_SENT =
+            HiveMQMetric.valueOf("com.hivemq.last-will-testaments.sent", Counter.class);
```
Comment (SgtSilvio) [inline src/main/java/com/hivemq/metrics/HiveMQMetrics.java]:
```suggestion
            HiveMQMetric.valueOf("com.hivemq.will.sent.count", Counter.class);
```

File: src/main/java/com/hivemq/metrics/HiveMQMetrics.java [hunk 3]
```diff
@@ -133,5 +129,17 @@
      */
     public static final HiveMQMetric<Gauge<Number>> CLIENT_SESSIONS_CURRENT =
             HiveMQMetric.gaugeValue("com.hivemq.sessions.overall.current");
+
+    /**
+     * represents a {@link Counter}, which measures the count of last will testaments.
+     */
+    public static final HiveMQMetric<Counter> LWT_CURRENT =
```
Comment (schaebo) [inline src/main/java/com/hivemq/metrics/HiveMQMetrics.java]: Should be named WILL_COUNT

File: src/main/java/com/hivemq/metrics/HiveMQMetrics.java [hunk 4]
```diff
@@ -133,5 +129,17 @@
...
+            HiveMQMetric.valueOf("com.hivemq.will.count", Counter.class);
+
+    /**
+     * represents a {@link Counter}, which measures the count of last will testaments spread to clients.
+     */
+    public static final HiveMQMetric<Counter> LWT_SENT =
```
Comment (schaebo) [inline src/main/java/com/hivemq/metrics/HiveMQMetrics.java]: Should be named WILL_SENT

File: src/main/java/com/hivemq/mqtt/handler/connect/ConnectPersistenceUpdateHandler.java [hunk 5]
```diff
@@ -143,6 +147,7 @@ private void sendClientDisconnect(final Channel channel, final String clientId,
                     public void onSuccess(@Nullable final Void result) {
                         if (!channel.attr(ChannelAttributes.TAKEN_OVER).get()) {
                             channelPersistence.remove(clientId);
+                            metricsHolder.getLastWillTestamentSentCounter().inc();
```
Comment (schaebo) [inline src/main/java/com/hivemq/mqtt/handler/connect/ConnectPersistenceUpdateHandler.java]: This does not mean that a will has been sent, it only means the client has been disconnected at the persistence level. It does also count disconnects without will. Better count the metric in com.hivemq.persistence.clientsession.PendingWillMessages#sendWill

**Pull request closed without merging**